src/core/EntryAttributes.cpp

```diff
@@ -69,6 +69,21 @@ bool EntryAttributes::isProtected(const QString& key) const
     return m_protectedAttributes.contains(key);
 }
 
+bool EntryAttributes::isReference(const QString& key) const
+{
+    if (!m_attributes.contains(key)) {
+        Q_ASSERT(false);
+        return false;
+    }
+
+    QString data = value(key);
+    QRegExp referenceRegExp("\\{REF:([TUPAN])@I:([^}]+)\\}", Qt::CaseInsensitive, QRegExp::RegExp2);
```

Comment:
I recommend creating this regexp object in the constructor and just using it. Creating it for every function call is wasteful since it is not dynamic. This function can be called many times in the course of searching.